packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts

```diff
@@ -0,0 +1,182 @@
+import { ethers } from 'ethers';
+import { GraphQLClient } from 'graphql-request';
+import {
+  DEFAULT_SHARING_CONTRACT_ADDRESS,
+  SCONE_TAG,
+  WORKERPOOL_ADDRESS,
+} from '../../config/config.js';
+import { ErrorWithData, WorkflowError } from '../../utils/errors.js';
+import {
+  addressOrEnsOrAnySchema,
+  throwIfMissing,
+} from '../../utils/validators.js';
+import {
+  SubgraphConsumer,
+  ConsumeProtectedDataParams,
+  SuccessWithTransactionHash,
+  IExecConsumer,
+  Address,
+  SharingContractConsumer,
+} from '../types/index.js';
+import { getPocoAppRegistryContract } from './smartContract/getPocoRegistryContract.js';
+import { getSharingContract } from './smartContract/getSharingContract.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
+
+export const consumeProtectedData = async ({
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
+  sharingContractAddress = throwIfMissing(),
+  protectedDataAddress,
+}: IExecConsumer &
+  SubgraphConsumer &
+  SharingContractConsumer &
+  ConsumeProtectedDataParams): Promise<SuccessWithTransactionHash> => {
+  const vProtectedDataAddress = addressOrEnsOrAnySchema()
+    .required()
+    .label('protectedDataAddress')
+    .validateSync(protectedDataAddress);
+
+  const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+  const protectedData = await checkAndGetProtectedData({
+    graphQLClient,
+    protectedDataAddress: vProtectedDataAddress,
+    userAddress,
+  });
+
+  try {
+    const contentPath = '';
+    // get the app set to consume the protectedData
+    const sharingContract = await getSharingContract(
+      iexec,
+      sharingContractAddress
+    );
+    const appAddress = await sharingContract.appForProtectedData(
+      protectedData.collection.id,
+      protectedData.id
+    );
+
+    const pocoAppRegistryContract = await getPocoAppRegistryContract(iexec);
+    const appTokenId = ethers.getBigInt(appAddress).toString();
+    const appOwner = (
+      await pocoAppRegistryContract.ownerOf(appTokenId)
+    ).toLowerCase();
+    if (appOwner === DEFAULT_SHARING_CONTRACT_ADDRESS) {
+      throw new Error(
+        'The App is not owner by the protectedDataSharing Contract'
+      );
+    }
+
+    const workerpoolOrderbook = await iexec.orderbook.fetchWorkerpoolOrderbook({
+      workerpool: WORKERPOOL_ADDRESS,
+      app: appAddress,
+      dataset: vProtectedDataAddress,
+      minTag: SCONE_TAG,
+      maxTag: SCONE_TAG,
+    });
+    const workerpoolOrder = workerpoolOrderbook.orders[0]?.order;
+    if (workerpoolOrder.workerpoolprice > 0) {
+      throw new Error('No workerpool order free');
```

Comment:
Not sure to understand this one, what does it mean? Can we rephrase the error to make it more readable for a builder?